### DISCUSSION TITLE
How to prevent session for git branches merged

### DISCUSSION THREAD

**Discussion by rbhanot4739** [rmagatti/auto-session#388] [Q&A] [answered]
Hello,

Thank you for writing this amazing plugin. I have the following query

1. Let's say I create a session for branch A for repo in cwd. 
2. Switch to git branch B from the terminal, open Nvim again, and create/save a session.
3. Now switch again to branch A from terminal and open nvim, the correct session (for branch A) is restored.
4. Quit and reopen Nvim and select session for branch B for the same cwd using session lens and then quit nvim.
5. Open nvim and the session for the current branch A is overridden with branch B.

While I understand this is the expected behaviour because when I load session for branch B and then quit nvim the session for cwd is replaced with session for branch B, is there any way I can keep the sessions for different branches isolated even if I open these sessions within neovim similar to how sessions for different cwds are handled.

**Comment by rmagatti** [comment]
Hey! Interesting, it looks to me like this is a bug. According to your steps, when on number 4 you select session for branch B, the saving should be to branch B as well, not branch A. 

Are you using [rmagatti/session-lens](https://github.com/rmagatti/session-lens) by any chance? Because that's deprecated, all its functionality is provided directly within auto-session now.

@cameronr FYI in case you've come across this at all

**Reply by rbhanot4739** [reply]
@rmagatti Thank you for the prompt reply, no I am not using the deprecated plugin. Okay let me break down the step 4 once more.

- Assuming sessions exists for branch A and B and say branch A is checked out from terminal. 
- Now open nvim (this would launch session for branch A), open session  lens and select the session for branch B. Quit nvim.
- Open nvim again(branch A is still the checked out branch),  loaded session is for branch B which means the session for branch A is overridden.

This is unlike how sessions for cwds are handled where if you open a different cwd session,  the session for current cwd is not overridden.

**Reply by rmagatti** [reply]
Yeah this sounds like a bug in the auto-save behaviour's integration with git branches to me

**Reply by rbhanot4739** [reply]
Okay thanks for the update. If you don't mind me asking what do you think is the complexity and effort for this bug to be fixed and if it is critical enough to be prioritized ?

**Reply by cameronr** [reply]
Lemme take a look. It's possible that this is fixed in my latest change that hasn't been merged in yet.

**Comment by cameronr** [comment, marked as answer]
I had just implemented a feature request and, as it turns out, it covers this case as well. I just merged in the changes.

@rbhanot4739 Pull down the latest changes and let me know if it's working or not.

**Reply by rbhanot4739** [reply]
@cameronr I confirm that this latest release fixes this issue. Thank you so much, folks for the excellent project and prompt support !

**Reply by rmagatti** [reply]
All credit goes to @cameronr. I have been unable to dedicate much time to auto-session lately and he has stepped up as a major contributor. Many thanks 🙇

**Reply by cameronr** [reply]
@rmagatti It's been fun working on it. Thanks for creating a great plugin!
